tests/flytekit/unit/core/test_dataclass.py

```diff
@@ -882,3 +883,33 @@ class NestedFlyteTypesWithDataClassJson(DataClassJsonMixin):
     transformer = DataclassTransformer()
     lt = transformer.get_literal_type(NestedFlyteTypesWithDataClassJson)
     assert lt.metadata is not None
+@pytest.mark.skipif(sys.version_info < (3, 10), reason="Requires Python 3.10 or higher")
+def test_numpy_import_issue_from_flyte_schema_in_dataclass():
+    from dataclasses import dataclass
+
+    from flytekit import task, workflow
```

Comment:
I added this test because in `flytekit==1.12.2` and `python==3.11`, it will fail.

@kumare3 and I traced the code and found that it's due to numpy import error.